cibyl/sources/jenkins_job_builder.py

```diff
@@ -44,49 +42,55 @@ def __init__(self, repos: dict = None,
                  enabled: bool = True, priority: int = 0,
                  name: str = "jenkins_job_builder",
                  driver: str = "jenkins_job_builder"):
-        """Create a client to talk to a jenkins job definitions instance.
 
-        :param repos: A dictionary of repositories to clone
-        :type repos: dict
-        """
         super().__init__(name=name, repos=repos, driver=driver,
                          enabled=enabled, priority=priority)
 
     def _generate_xml(self):
         """Use tox to generate jenkins job xml files."""
         for repo in self.repos:
-            subprocess.run(["tox",  "-e", "jobs"], check=True,
+            subprocess.run(["tox", "-e", "jobs"], check=True,
                            cwd=repo.get('dest'))
 
     @speed_index({'base': 1})
     def get_jobs(self, **kwargs):
-        """Get all jobs from jenkins server.
-
-        :returns: All jobs from jenkins server, as dictionaries of _class,
-        name, fullname, url, color
-        :rtype: list
+        """Get jobs from a given repo
+            :returns: container of Job objects extracted from JJB generated
+                      xml files
+            :rtype: :class:`AttributeDictValue`
         """
         all_jobs = {}
         for repo in self.repos:
-            all_jobs.update(self.get_jobs_from_repo(repo, **kwargs))
+
+            all_jobs.update(
+                self.get_jobs_from_repo(frozenset(repo.items()), **kwargs))
         return AttributeDictValue("jobs", attr_type=Job, value=all_jobs)
 
     def get_jobs_from_repo(self, repo, **kwargs):
-        """Get all jobs from a given repo
+        """Get jobs from a given repo
 
-        :param repo: repo information like path and url
-        :type repo: dict
-        :returns: All jobs from the given repository
-        :rtype: dict
+            :returns: container of Job objects extracted from JJB generated
+                      xml files
+            :rtype: :class:`AttributeDictValue`
         """
+        repo = dict(repo)  # convert set 2 dictionary
```

Comment:
It seems that get_jobs is converting the repo dict to a set and then here converting back to a dictionary. Is there any reason for this conversions or could we just pass the original repo dictionary?

Reply:
> It seems that get_jobs is converting the repo dict to a set and then here converting back to a dictionary. Is there any reason for this conversions or could we just pass the original repo dictionary?

Done